inspire/modules/authors/receivers.py

```diff
@@ -1,34 +1,48 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of INSPIRE.
-# Copyright (C) 2014, 2015 CERN.
+# Copyright (C) 2015 CERN.
 #
-# INSPIRE is free software: you can redistribute it and/or modify
-# it under the terms of the GNU General Public License as published by
-# the Free Software Foundation, either version 3 of the License, or
-# (at your option) any later version.
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
 #
-# INSPIRE is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
 #
 # You should have received a copy of the GNU General Public License
-# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
-#
-# In applying this licence, CERN does not waive the privileges and immunities
-# granted to it by virtue of its status as an Intergovernmental Organization
-# or submit itself to any jurisdiction.
-
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
 
 import uuid
 
 from invenio_records.signals import (
+    before_record_index,
     before_record_insert,
     before_record_update,
 )
 
 
+@before_record_index.connect
+def generate_name_variatons(recid, json):
+    """Populates a json record before indexing it to elastic.
+    Adds a field for all the possible variations of an authors name
+
+    :param recid: The id of the record that is going to be indexed.
+    :param json: The json representatio of the record that is going to be indexed.
+    """
+    from inspire.modules.authors.utils import author_tokenize
+    authors = json.get("authors")
+    if authors is not None:
```

Comment:
Actually you might want to just write:
`if authors:`

since `authors` could be an empty list.